packages/go/src/symbols/named-type.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Children,
+  Namekey,
+  OutputSpace,
+  Refkey,
+  track,
+  TrackOpTypes,
+  trigger,
+  TriggerOpTypes,
+} from "@alloy-js/core";
+import { GoSymbol, GoSymbolOptions } from "./go.js";
+
+// represents a symbol from a .go file. Struct, interface, etc.
+export type NamedTypeTypeKind =
+  | "import"
+  | "function"
+  | "var"
+  | "type"
+  | "struct"
+  | "struct-member"
+  | "interface"
+  | "interface-member"
+  | "package";
+export type NamedTypeSymbolKind = "named-type" | "package";
+
+/**
+ * Information for a Go generic type parameter.
+ */
+export interface TypeParameterProps {
```

Comment:
Move this with the component?

Reply:
If I leave TypeParameterProps in `typeparameters.tsx` then `pnpm build` complains that `The symbol "TypeParameterProps_2" needs to be exported by the entry point index.d.ts` in `named-type.ts`. I'm not sure why that would error since `TypeParameterProps` is exported from `index.ts`